app/executor.go

```diff
@@ -188,40 +192,65 @@ func (ms stmMultiStoreWrapper) GetObjKVStore(key storetypes.StoreKey) storetypes
 
 // preEstimates returns a static estimation of the written keys for each transaction.
 // NOTE: make sure it sync with the latest sdk logic when sdk upgrade.
-func preEstimates(txs []sdk.Tx, authStore, bankStore int, evmDenom string) map[int]blockstm.MultiLocations {
-	estimates := make(map[int]blockstm.MultiLocations, len(txs))
-	for i, tx := range txs {
-		feeTx, ok := tx.(sdk.FeeTx)
-		if !ok {
-			continue
-		}
-		feePayer := sdk.AccAddress(feeTx.FeePayer())
+func preEstimates(txs [][]byte, workers, authStore, bankStore int, evmDenom string, txDecoder sdk.TxDecoder) ([]sdk.Tx, []blockstm.MultiLocations) {
+	memTxs := make([]sdk.Tx, len(txs))
+	estimates := make([]blockstm.MultiLocations, len(txs))
+
+	job := func(start, end int) {
+		for i := start; i < end; i++ {
+			rawTx := txs[i]
+			tx, err := txDecoder(rawTx)
+			if err != nil {
+				continue
+			}
+			memTxs[i] = tx
 
-		// account key
-		accKey, err := collections.EncodeKeyWithPrefix(
-			authtypes.AddressStoreKeyPrefix,
-			sdk.AccAddressKey,
-			feePayer,
-		)
-		if err != nil {
-			continue
-		}
+			feeTx, ok := tx.(sdk.FeeTx)
+			if !ok {
+				continue
+			}
+			feePayer := sdk.AccAddress(feeTx.FeePayer())
+
+			// account key
+			accKey, err := collections.EncodeKeyWithPrefix(
+				authtypes.AddressStoreKeyPrefix,
+				sdk.AccAddressKey,
+				feePayer,
+			)
+			if err != nil {
```

Comment:
do we need log out these error

Reply:
I think it's fine, if the tx is invalid, the execution will return err anyway.